pkg/scheduler/framework/plugins/podtopologyspread/scoring.go

```diff
@@ -104,7 +107,7 @@ func (pl *PodTopologySpread) PreScore(
 	}
 
 	state := &preScoreState{
-		NodeNameSet:             sets.String{},
+		NodeNameSet:             make(sets.String, len(filteredNodes)),
```

Comment:
Just for curiosity: how much percentage does it help in the performance boost?

Reply:
The benchmark for 100 nodes remains at 0.3ms and goes up to 2.1ms for 1000 nodes.
The profile shows 10% CPU usage from `sets.String.Insert`, dominated by growing the hash table, which is not negligible.
I think it's fair to assume that if a user sets spreading, they expect all nodes to have the topology keys.